inventory/src/components/KitIndex/KitAdd.js

```diff
@@ -0,0 +1,218 @@
+import AddIcon from '@mui/icons-material/Add';
+import CancelIcon from '@mui/icons-material/Cancel';
+import SendIcon from '@mui/icons-material/Send';
+import { LoadingButton } from '@mui/lab';
+import {
+  Box,
+  FormControl,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Modal,
+  Select,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { useFormik } from 'formik';
+import React, { useState } from 'react';
+
+import { useCreateKit } from '../../hooks/mutations/useCreateKit';
+import { useKitRecipe } from '../../hooks/queries/useKitRecipe';
+
+import { BlueprintSelect } from './BlueprintSelect';
+import { kitAddValidation } from './schema';
+
+export const KitAdd = ({ blueprints }) => {
+  const [open, setOpen] = useState(false);
+  const [selectedBlueprint, setSelectedBlueprint] = useState('');
+  const { data: kitRecipeData } = useKitRecipe(selectedBlueprint);
+  const { mutate, isLoading } = useCreateKit({});
+
+  const formik = useFormik({
+    initialValues: {
+      name: '',
+      blueprint: '',
+      kitContent: {},
+      itemTotalQty: {},
+    },
+    validationSchema: kitAddValidation,
+    validateOnChange: false,
+    onSubmit: () => {
+      const content = Object.values(formik.values.kitContent)
+        .map((v) => Object.values(v))
+        .flat();
```

Comment:
When submitting a form with no values filled in (kit is empty), there will be an unhandled error